app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala

```diff
@@ -16,64 +16,87 @@
 
 package uk.gov.hmrc.apiplatformjobs.repository
 
-import akka.stream.Materializer
-import org.joda.time.DateTime
-import play.api.libs.json.{Format, Json}
-import play.modules.reactivemongo.ReactiveMongoComponent
-import reactivemongo.api.indexes.Index
-import reactivemongo.api.indexes.IndexType.Ascending
-import reactivemongo.bson.BSONObjectID
+import java.time.Clock
+import org.mongodb.scala.model.InsertOneModel
+import org.mongodb.scala.bson.collection.immutable.Document
+import org.mongodb.scala.model.Filters.{and, equal, lte}
+import org.mongodb.scala.model.{FindOneAndUpdateOptions, IndexModel, IndexOptions}
+import org.mongodb.scala.model.Indexes.ascending
 import uk.gov.hmrc.apiplatformjobs.models.Environment.Environment
 import uk.gov.hmrc.apiplatformjobs.models.{MongoFormat, UnusedApplication}
-import uk.gov.hmrc.mongo.ReactiveRepository
-import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.formats.MongoJavatimeFormats
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
 
+import java.time.LocalDateTime
 import java.util.UUID
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
+import org.mongodb.scala.model.Updates
+import com.mongodb.client.model.ReturnDocument
 
 @Singleton
-class UnusedApplicationsRepository @Inject()(mongo: ReactiveMongoComponent)(implicit val mat: Materializer, val ec: ExecutionContext)
-  extends ReactiveRepository[UnusedApplication, BSONObjectID]("unusedApplications", mongo.mongoConnector.db,
-    MongoFormat.unusedApplicationFormat, ReactiveMongoFormats.objectIdFormats) {
+class UnusedApplicationsRepository @Inject()(mongo: MongoComponent, val clock: Clock)(implicit val ec: ExecutionContext)
+  extends PlayMongoRepository[UnusedApplication](
+    collectionName = "unusedApplications",
+    mongoComponent = mongo,
+    domainFormat = MongoFormat.unusedApplicationFormat,
+    indexes = Seq(
+      IndexModel(ascending("environment", "applicationId"), IndexOptions()
+        .name("applicationIdIndex")
+        .unique(true)
+        .background(true)
+      ),
+      IndexModel(ascending("environment", "scheduledNotificationDates"), IndexOptions()
+        .name("scheduledNotificationDatesIndex")
+        .unique(false)
+        .background(true)
+      ),
+      IndexModel(ascending("environment", "scheduledDeletionDate"), IndexOptions()
+        .name("scheduledDeletionDateIndex")
+        .unique(false)
+        .background(true)
+      )
+    ),
+    replaceIndexes = true
+  ) with MongoJavatimeFormats.Implicits {
```

Comment:
Does this implicit do anything?

Reply:
Yes I think it is used in date queries i.e. `Codecs.toBson(deletionDate)`